kedro-airflow/kedro_airflow/grouping.py

```diff
@@ -25,74 +20,88 @@ def get_memory_datasets(catalog: DataCatalog, pipeline: Pipeline) -> set[str]:
     }
 
 
-def node_sequence_name(node_sequence: list[Node]) -> str:
-    return "_".join([node.name for node in node_sequence])
+def _build_adjacency_matrix(
+    catalog: DataCatalog, pipeline: Pipeline
+) -> tuple[dict[str, set], dict[str, set]]:
+    """
+    Builds adjacency matrix (adj_matrix) to search connected components - undirected graph,
+    and adjacency matrix (parents) to retrieve connections between new components using on
+    initial kedro topological sort - directed graph.
+    """
+    memory_datasets = get_memory_datasets(catalog, pipeline)
+
+    adj_matrix: dict[str, set] = {node.name: set() for node in pipeline.nodes}
+    parents: dict[str, set] = {node.name: set() for node in pipeline.nodes}
+    output_to_node = {
+        node_output: node for node in pipeline.nodes for node_output in node.outputs
+    }
+
+    for node in pipeline.nodes:
+        for node_input in node.inputs:
+            if node_input in output_to_node:
+                if node_input in memory_datasets:
+                    adj_matrix[node.name].add(output_to_node[node_input].name)
+                    adj_matrix[output_to_node[node_input].name].add(node.name)
+                parents[output_to_node[node_input].name].add(node.name)
```

Comment:
To me, it seems like that's a `children` dictionary, because the values are children, as I understand it.

Reply:
Yeah, a key is a parent's name, while value is the children's list. Will rename it to `parent_to_children` for clarity. Thank you!

Reply:
Done